C#: Error in generated code for WIT variant containing `f32`

_See kaivol/wit-bindgen#1 for a reproduction of the error._

Consider the following WIT variant:

````wit
variant v1 {
    d(string),
    g(u32),
    h(f32),
}
````

For a function `v1-arg: func(x: v1)`, the generated C# code looks like this: 
````csharp
[UnmanagedCallersOnly(EntryPoint = "foo:foo/variants#v1-arg")]
public static unsafe void wasmExportV1Arg(int p0, nint p1, int p2) {
    ...
    // case g
    lifted = global::MyWorldWorld.wit.exports.foo.foo.IVariants.V1.g(unchecked((uint)(p1)));
    ...
    // case h
    lifted = global::MyWorldWorld.wit.exports.foo.foo.IVariants.V1.h(BitConverter.Int32BitsToSingle(p1));
    ...
}
````
Compilation fails with `error CS1503: Argument 1: cannot convert from 'nint' to 'int'`, because `Int32BitsToSingle` expects a 32-bit `int`, but `p1` is a pointer-sized `nint` (even though `nint` should also be 32-bit wide because WASM pointers are always 32-bit, ignoring the Memory64 proposal).

--- a/crates/csharp/src/lib.rs
+++ b/crates/csharp/src/lib.rs
@@ -3227,7 +3227,7 @@ fn dotnet_aligned_array(array_size: usize, required_alignment: usize) -> (usize,
 
 fn perform_cast(op: &String, cast: &Bitcast) -> String {
     match cast {
-        Bitcast::I32ToF32 => format!("BitConverter.Int32BitsToSingle({op})"),
+        Bitcast::I32ToF32 => format!("BitConverter.Int32BitsToSingle((int){op})"),
         Bitcast::I64ToF32 => format!("BitConverter.Int32BitsToSingle((int){op})"),
         Bitcast::F32ToI32 => format!("BitConverter.SingleToInt32Bits({op})"),
         Bitcast::F32ToI64 => format!("BitConverter.SingleToInt32Bits({op})"),

--- a/tests/codegen/variants.wit
+++ b/tests/codegen/variants.wit
@@ -21,6 +21,7 @@ interface variants {
       e(empty),
       f,
       g(u32),
+      h(f32),
   }
 
   v1-arg: func(x: v1);

